LIS/lis-test
# Add hv-sock test case: HV_Sock_Basic

This test covers basic socket connection and data transmission using
hv-sock.

Part I: Client app on vm connects to server app on host
Part II: Client app on host connects to server app on vm

Manual test steps:

Setup:

1. Add register key for GuestCommunicationService on host:

    On host, run "regedit", find open the path:
    HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\
    Virtualization\GuestCommunicationServices\

    If the key "00000808-facb-11e6-bd58-64006a7986d3" does not exists,
    add it, set value "HV Socket Service" for the key

2. Load module on guest linux:
    \# modprobe hv-sock

Test Part I:

1. Run server_on_host.exe on host
2. Run client_on_vm on guest linux
3. Check output on both sides, connected successfully without error.

Test Part II:

1. Get VM GUID in powershell:
    > Get-VM -Name "Your VM Name" |fl name, id
2. Run server_on_vm on guest linux
3. Run client_on_host.exe on host with argument VM GUID:
    > client_on_host.exe
3. Check output on both sides, connected successfully without error.
